zad5.js

```diff
@@ -0,0 +1,99 @@
+const people = [
+  {
+    firstName: "Bartolomeo",
+    lastName: "Lozano",
+    nickname: "Rabona",
+    introduceYourself: introduceYourself,
+    getFavouriteColor: getFavouriteColor,
+  },
+  {
+    firstName: "Mateo",
+    lastName: "Loza",
+    nickname: "Tamazo",
+    introduceYourself: introduceYourself,
+    getFavouriteColor: getFavouriteColor,
+  },
+  {
+    firstName: "Kasia",
+    lastName: "Kowalska",
+    nickname: "Sakaks",
+    introduceYourself: introduceYourself,
+    getFavouriteColor: getFavouriteColor,
+  },
+  {
+    firstName: "Jack",
+    lastName: "Payne",
+    nickname: "Cajeny",
+    introduceYourself: introduceYourself,
+    getFavouriteColor: getFavouriteColor,
+  },
+];
+
+const colors = ["red", "green", "yellow", "blue", "pink", "orange"];
+
+function getFavouriteColor(person, number) {
+  const stringSum =
+    person.firstName.length + person.lastName.length + person.nickname.length;
+
+  const whatColor = (stringSum - number) % colors.length;
+
+  if (number == null) {
+    return colors[5];
+  } else if (number < 1) {
+    return "Podałeś/łaś za małą liczbę, liczba nie może być mniejsza niż 1";
+  } else if (number > 30) {
+    return "Podałeś/łaś za dużą liczbę, liczba nie może być większa niż 30";
+  } else {
+    return colors[whatColor];
+  }
+}
+
+function introduceYourself() {
+  return console.log(
+    `Cześć jestem ${this.firstName} ${this.lastName}, ale w szkole mówią na mnie [${this.nickname}]`
+  );
+}
+
+const randomNum = Math.floor(Math.random() * 101);
+
+function isElite(num) {
+  const isPrime = (num) => {
```

Comment:
Jeśli używasz tej samej nazwy parametru to nie musisz go w tym miejscu znów deklarować. Funkcja i tak go rozpozna zgodnie z jej zakresem.